### DISCUSSION TITLE
Ord for Bools

### DISCUSSION THREAD

**Discussion by oxinabox** [google-research/dex-lang#325] [Ideas]
consider
```
>=> :p True < False
Type error:Couldn't synthesize a class dictionary for: (Ord Bool)

:p True < False
        ^^
```

Though the main reason i wanted this was that `findfirst` and `findlast` could then be just be `argmin` and `argmax` when applied to a table of booleans.
Perhaps better would be to implement those directly

**Comment by apaszke** [comment]
I have mixed feelings about adding ordering to booleans 😕 On the other hand we already allow those as index sets, but I've started wondering if that wasn't a mistake.

**Comment by oxinabox** [comment]
> I have mixed feelings about adding ordering to booleans 

Agreed.

You could have `Bool` and `Int1` as seperate types.

**Comment by apaszke** [comment]
Yes, that sounds reasonable. Or just use `Fin 2`

**Comment by dan-zheng** [comment]
I would also lean towards not adding an `Ord Bool` instance unless there's an unsolved motivating use case (where `Fin 2` doesn't work or conversion is too unergonomic).

Some precedent: [Haskell has an `Ord Bool` instance](https://hackage.haskell.org/package/base-4.14.1.0/docs/Data-Ord.html#i:Ord) while [Swift does not](https://forums.swift.org/t/why-bool-doesnt-conform-to-comparable/10698).

I'll move this to Discussions now, since this is more like a discussion than an issue.
